statistics/table.go

```diff
@@ -414,25 +420,57 @@ func getPseudoRowCountByColumnRanges(sc *stmtctx.StatementContext, tableRowCount
 	return rowCount, nil
 }
 
-func getPseudoRowCountByIntRanges(intRanges []ranger.IntColumnRange, tableRowCount float64) float64 {
+func getPseudoRowCountBySignedIntRanges(intRanges []*ranger.NewRange, tableRowCount float64) float64 {
```

Comment:
This function is complex and similar with uint version. Is there a good way to simplify it?

Reply:
We can simplify this later. Datum's Compare method needs to handle error, it's not so good to use datum to merge this two function.